cats/views.py

```diff
@@ -0,0 +1,154 @@
+"""
+Views used in cat app
+"""
+
+import datetime
+
+from django.contrib.auth.decorators import login_required
+from django.core.mail import send_mail
+from django.core.paginator import Paginator
+from django.shortcuts import render, get_object_or_404, redirect
+from django.template.loader import render_to_string
+from django.urls import reverse
+from django.views.generic import TemplateView, ListView, DetailView
+
+from cats.forms import RentalForm, SearchForm
+from cats.models import Cat, Species, Breed, Rental
+
+
+class IndexView(TemplateView):
+    """Simple index view"""
+    template_name = 'cats/index.html'
+
+
+class AboutView(TemplateView):
+    """Simple about view"""
+    template_name = 'cats/about.html'
+
+
+class SpeciesListView(ListView):
+    """View to choose a Species"""
+    model = Species
+    template_name = 'cats/species.html'
+
+
+def explore_list(request):
+    """Lists Cats from all kinds of Species"""
+    cats = None
+    page_obj = None
+    search_form = SearchForm(request.GET or None)
+
+    """
+    If form is valid, show list.
+    If form is not valid, SearchForm will show proper hint
+    """
+    if search_form.is_valid():
+        date_from = search_form.cleaned_data['date_from']
+        date_to = search_form.cleaned_data['date_to']
+        cats = Cat.objects.filter_by_dates(date_from, date_to)
+
+        paginator = Paginator(cats, 5)
+        page_number = request.GET.get('page')
+        page_obj = paginator.get_page(page_number)
+
+    context = {'cats_list': cats, 'search_form': search_form, 'page_obj': page_obj}
+    return render(request, 'cats/explore_list.html', context)
+
+
+def cats_list(request, species_id):
```

Comment:
Why not `FormView`?

Reply:
or even ListView. You should use Class Based views :D